zabbix_auto_config/processing.py

```diff
@@ -661,16 +681,25 @@ def do_update(self):
                     logging.debug("Going to add tags '%s' to host '%s'.", tags_to_add, zabbix_host["host"])
                 self.set_tags(zabbix_host, tags)
 
-            # Check the inventory
-            # inventory object lacks inventory_mode attr. in api >= 4.4
-            # if api ver >= 4.4
-            if "inventory_mode" in zabbix_host.keys():
-                if zabbix_host["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
-            # else api ver. < 4.4
-            else:
-                if zabbix_host["inventory"]["inventory_mode"] != "1":
-                    self.set_inventory(zabbix_host, 1)
+            if zabbix_host["inventory_mode"] != "1":
```

Comment:
I just noticed that this is a string comparrison (because the API returns a string). According to the Zabbix docs, this should be an int. So Zabbix is bugged. Previously Zabbix has suddenly and silently fixed these kind of issues. I believe that we should convert this to an int along with the other fixes here:

https://github.com/unioslo/zabbix-auto-config/blob/56e41c00be245ae8fe8e3e71ecf922628f45f3f3/zabbix_auto_config/processing.py#L607-L612

Reply:
IIRC _all_ values retrieved from the Zabbix API are strings. Perhaps a deliberate decision to prevent sudden type changes as apparently occured in the past?

Reply:
I'm not sure what the best approach is. The docs are bad. I see now that they mention this [in the "Attention" box here](https://www.zabbix.com/documentation/current/manual/api/reference_commentary). Their statement is false though. They also return objects.

I think we should be consistent at least? Either we convert the ints to actual ints, or we don't? But this could maybe be filed as its own issue?

Reply:
+1 for consistency.  Casting to native types makes a lot of sense.